### PR TITLE
chore: add CI/CD pipeline, Issue templates, contribution md

### PR DESCRIPTION
**Pull request by cialese**

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Use Node.js 20
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20.x'
+      - name: Set up .npmrc
+        run: echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > .npmrc
+      - name: Install Dependencies
+        run: npm install
```
Comment (osbornm) [inline .github/workflows/release.yaml]: we should use a cahce for these

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      - name: Linter Check
+        run: npm run lint
+      - name: Run Tests
+        run: npm run test
+      - name: Run Build
+        run: npm run build
```
Comment (osbornm) [inline .github/workflows/release.yaml]: these can run in parrallel

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+on:
+  workflow_dispatch:
+  push:
+    branches: [main]
+
+permissions:
+  contents: write
+  issues: write
+  pull-requests: write
+  id-token: write
```
Comment (osbornm) [inline .github/workflows/release.yaml:10]: I dont think you need all these permissions

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      - name: Linter Check
+        run: npm run lint
+      - name: Run Tests
+        run: npm run test
+      - name: Run Build
+        run: npm run build
+      - name: Publish to npm
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+        run: npm publish --access public
```
Comment (osbornm) [inline .github/workflows/release.yaml]: the release process should handle the package version. consider using symantic releases like we did with hyphen components

File: .github/workflows/pull-request.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        uses: actions/checkout@v4
+
+      # Setup Node
+      - name: Use Node.js 20
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      - name: Install Dependencies
+        run: npm install
```
Comment (osbornm) [inline .github/workflows/pull-request.yaml]: use a cache

File: .github/workflows/pull-request.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+      - name: Linter Check
+        run: npm run lint
+
+      - name: Run Tests
+        run: npm run test
+
+      - name: Run Build
+        run: npm run build
```
Comment (osbornm) [inline .github/workflows/pull-request.yaml]: run in parallel

**Review by osbornm — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yaml (hunk 4 above)
Comment (cialese) [reply .github/workflows/release.yaml]: @osbornm  will the release be made automatically when a push is made to the main branch, or will it be done manually using GitHub Releases like the CLI?

Comment (osbornm) [reply .github/workflows/release.yaml]: I'd like to handle this exactly like hyphen components does. Please look at how it handles releases and how it updates package.json as well.

Comment (osbornm) [reply .github/workflows/release.yaml]: another good example https://github.com/Hyphen/nucleus?tab=readme-ov-file#releases

File: .github/workflows/pull-request.yaml [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+permissions:
+  id-token: write
+  contents: read
+  packages: read
+  repository-projects: read
+
+env:
+  GH_PACKAGES_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (osbornm) [inline .github/workflows/pull-request.yaml]: not needed

File: .github/workflows/pull-request.yaml [hunk 8]
```diff
@@ -0,0 +1,87 @@
+name: PR Check
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [main]
```
Comment (osbornm) [inline .github/workflows/pull-request.yaml]: These are the wrong events for a workflow named pull request.

File: .github/workflows/pull-request.yaml [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+# Cancel any jobs in-progress for the current branch
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  id-token: write
+  contents: read
+  packages: read
+  repository-projects: read
```
Comment (osbornm) [inline .github/workflows/pull-request.yaml]: Not sure you need all these permissions for a PR workflow. If you do can you explain why?

File: .github/workflows/release.yaml [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+  release:
+    name: Release Package
+    runs-on: ubuntu-latest
+    needs: [lint, test, build]
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '20.x'
+      - name: Release to Github Packages
```
Comment (osbornm) [inline .github/workflows/release.yaml]: we want to release to NPM not Github. This might just be left over copy and paste?

**Review by osbornm — CHANGES_REQUESTED** [review]

**Review by osbornm — APPROVED** [review]
This looks good, When you merge it first make sure the version in package.json is something like `1.0.0-beta.1` and then merge it and make sure everything works as planned. Follow up with any bug fixes to the pipeline as needed.

**Pull request merged**